### PR TITLE
update and add examples to new APIs

### PR DESCRIPTION
**Pull request by tomerd**
motivation: keep code up to date with API changes

changes:
* take advantage of @main where possible
* move the top level Sample code to the examples subdirectory
* extract a few examples form the "LambdaFunctions" directory (which is really a deployment demo) and move them to the top level Examples directory
*  rename "LambdaFunctions" examples as "Deployments" to make their intent clearer
* add a sample that demonstrates how to test a lambda now that SwiftPM can test executables directly
* update the test-sample docker setup to build & test th new samples
* fix a few typos and In/Out typealias left overs
* remove LinuxMain since its no longer required

### CODE REVIEW

**Comment by fabianfett** [comment]
What's the benefit of moving the String and Codable sample out of the main `Package.swift`?

**Comment by tomerd** [comment]
> What's the benefit of moving the String and Codable sample out of the main Package.swift?

its much less noisy. these are really just samples and so they should not be part of the main package. One area where this causes headache is that tools that scan packages (like package collections and package indexes) and cannot tell that these modules are "not important" so they show up as valid modules you may want to use, which his not the intent

File: Examples/LambdaFunctions/Sources/Benchmark/main.swift [hunk 1]
```diff
@@ -19,7 +19,6 @@ import NIO
 // use this example which is more performant.
 // `EventLoopLambdaHandler` does not offload the Lambda processing to a separate thread
 // while the closure-based handlers do.
```
Comment (stevapple) [inline Examples/LambdaFunctions/Sources/Benchmark/main.swift:21]: I think the description here should also be modified?

File: Examples/Simple/Testing/Sources/Lambda.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaRuntime
+import NIOCore
```
Comment (stevapple) [inline Examples/Simple/Testing/Sources/Lambda.swift:16]: Is `NIOCore` really necessary here (and in other `LambdaHandler` examples)?

**Review by stevapple — COMMENTED** [review]
Looks much neater!

File: Examples/Simple/String/Lambda.swift [hunk 3]
```diff
@@ -13,17 +13,20 @@
 //===----------------------------------------------------------------------===//
 
 import AWSLambdaRuntimeCore
```
Comment (stevapple) [inline Examples/Simple/String/Lambda.swift:15]: Encourage using `AWSLambdaRuntime` here?

File: Examples/LambdaFunctions/Sources/Benchmark/main.swift [hunk 4]
```diff
@@ -19,7 +19,6 @@ import NIO
 // use this example which is more performant.
```
Comment (stevapple) [inline Examples/LambdaFunctions/Sources/Benchmark/main.swift:19]: `NIOCore` is enough I think.

**Review by stevapple — COMMENTED** [review]
We shall encourage `AWSLambdaRuntimeCore` with `NIOCore` with `EventLoopLambdaHandler`, and `AWSLambdaRuntime` with `LambdaHandler`?

**Review by fabianfett — APPROVED** [review]
Looks great. Much cleaner! Thanks!

**Pull request merged**